tensorflow/tensorboard
# mesh: replace broken caches with direct lookups

Summary:
Fixes #2579, and fixes a few unreported bugs regarding run deletions.
These caches were broken, because the underlying data could change but
the caches were never updated or invalidated. The behavior that they
memoize over is not very expensive (a few shallow copies and parsing of
small proto messages), so we simply remove the caches in favor of direct
access. If this proves to be too expensive, then we can look into more
carefully reinstating the caches.

Test Plan:

 1. Run the mesh demo to generate data into `mesh_demo/run1` with tag
    `mesh1`.
 2. Start TensorBoard pointing at `mesh_demo`. Verify that the mesh
    appears.
 3. Generate test data into `mesh_demo/run1` with tag `mesh2`, then
    again into `mesh_demo/run2` with tag `mesh2`.
 4. Click TensorBoard’s refresh button, or reload the page, and note
    that the new meshes appear.
 5. Remove the `mesh_demo/run1` directory.
 6. Reload TensorBoard, and verify that only one mesh remains, under the
    correct run.

wchargin-branch: mesh-cache-remove
